s2a/src/main/java/io/grpc/s2a/S2AChannelCredentials.java

```diff
@@ -106,15 +142,26 @@ public Builder setLocalUid(String localUid) {
       return this;
     }
 
-    /** Sets the credentials to be used when connecting to the S2A. */
-    @CanIgnoreReturnValue
-    public Builder setS2AChannelCredentials(ChannelCredentials s2aChannelCredentials) {
-      this.s2aChannelCredentials = s2aChannelCredentials;
-      return this;
-    }
-
-    public ChannelCredentials build() {
+    public ChannelCredentials build() throws IOException {
       checkState(!isNullOrEmpty(s2aAddress), "S2A address must not be null or empty.");
+      ChannelCredentials s2aChannelCredentials;
+      if (useMtlsToS2A) {
+        checkState(!isNullOrEmpty(privateKeyPath), "privateKeyPath must not be null or empty.");
+        checkState(!isNullOrEmpty(certChainPath), "certChainPath must not be null or empty.");
+        checkState(!isNullOrEmpty(trustBundlePath), "trustBundlePath must not be null or empty.");
+
+        File privateKeyFile = new File(privateKeyPath);
+        File certChainFile = new File(certChainPath);
+        File trustBundleFile = new File(trustBundlePath);
+
```

Comment:
You should verify that the files exist so that you can provide a meaningful error message if they don't.

Reply:
Done in https://github.com/grpc/grpc-java/pull/11544/commits/2407bb6aae01bccbb7b3367feb521cc8ad962299